library/yaml/dump_spec.rb

```diff
@@ -1,17 +1,20 @@
 require_relative '../../spec_helper'
 require_relative 'fixtures/common'
 
-# TODO: WTF is this using a global?
 describe "YAML.dump" do
+  before :each do
+    @test_file = tmp("yaml_test_file")
+  end
+
   after :each do
-    rm_r $test_file
+    rm_r @test_file
   end
 
   it "converts an object to YAML and write result to io when io provided" do
-    File.open($test_file, 'w' ) do |io|
+    File.open(@test_file, 'w' ) do |io|
       YAML.dump( ['badger', 'elephant', 'tiger'], io )
     end
-    YAML.load_file($test_file).should == ['badger', 'elephant', 'tiger']
+    YAML.load_file(@test_file).should == ['badger', 'elephant', 'tiger']
   end
```

Comment:
**suggestion:** `$test_file` was used in one case only - it makes sense not to create and delete file for every other cases.